src/gui/src/devsettings_widgets.cpp

```diff
@@ -0,0 +1,1254 @@
+  box_rcd = new QComboBox(cw);
+  for (int i = 0; sm_rcd[i] > 0; i++)
+    box_rcd->addItem(QString::number(sm_rcd[i]) + "X");
+  box_wcd = new QComboBox(cw);
+  for (int i = 0; sm_wcd[i] > 0; i++)
+    box_wcd->addItem(QString::number(sm_wcd[i]) + "X");
+  box_rdvd = new QComboBox(cw);
+  for (int i = 0; sm_rdvd[i] > 0; i++)
+    box_rdvd->addItem(QString::number(sm_rdvd[i]) + "X");
+  box_wdvd = new QComboBox(cw);
+  //	for (int i=0; sm_wdvd[i]>0; i++)
+  //box_wdvd->addItem(QString::number(sm_wdvd[i]) + "X");
+
+  if (!(dev->cap_rd & DEVICE_DVD)) {
+    spdd->setEnabled(false);
+    box_rdvd->setEnabled(false);
+  }
+  box_wdvd->setEnabled(false);
+
+  // Access time
+  h_access = new QLabel(tr("Access Time"), cw);
+  h_access->setAlignment(Qt::AlignCenter);
+  h_access->setFrameStyle(QFrame::StyledPanel | QFrame::Sunken);
+
+  a_slow = new QRadioButton(tr("Slow"), cw);
+  a_fast = new QRadioButton(tr("Fast"), cw);
+  grp_access = new QButtonGroup(cw);
+  grp_access->addButton(a_slow, 0);
+  grp_access->addButton(a_fast, 1);
+  a_slow->setChecked(true);
+
+  // Tray speed
+  h_tray = new QLabel(tr("Tray Speed"), cw);
+  h_tray->setAlignment(Qt::AlignCenter);
+  h_tray->setFrameStyle(QFrame::StyledPanel | QFrame::Sunken);
+
+  l_load = new QLabel(tr("Load"), cw);
+  l_eject = new QLabel(tr("Eject"), cw);
+  s_load = new QSlider(Qt::Horizontal, cw);
+  s_load->setRange(0, 80);
+  s_eject = new QSlider(Qt::Horizontal, cw);
+  s_eject->setRange(0, 80);
+
+  // Buttons
+  ck_perm = new QCheckBox(tr("Permanent"), this);
+  pb_save = new QPushButton(QIcon(":images/ok.png"), tr("Save"), this);
+  pb_save->setMinimumWidth(100);
+
+  // Layouts
+  layoutm = new QVBoxLayout(this);
+  layoutm->setMargin(0);
+  layoutm->setSpacing(3);
+  layoutm->addWidget(cw);
+
+  layout = new QGridLayout(cw);
+  layout->setMargin(3);
+  layout->setSpacing(3);
+
+  layout->addWidget(h_speed, 0, 0, 1, 4);
+
+  layout->addWidget(spdc, 1, 1, 1, 2);
+  layout->addWidget(spdd, 1, 3);
+  layout->addWidget(spdr, 2, 0);
+  layout->addWidget(box_rcd, 2, 1, 1, 2);
+  layout->addWidget(box_rdvd, 2, 3);
+  layout->addWidget(spdw, 3, 0);
+  layout->addWidget(box_wcd, 3, 1, 1, 2);
+  layout->addWidget(box_wdvd, 3, 3);
+
+  layout->addWidget(h_access, 4, 0, 1, 4);
+
+  layout->addWidget(a_slow, 5, 0, 1, 2);
+  layout->addWidget(a_fast, 5, 2, 1, 2);
+
+  layout->addWidget(h_tray, 6, 0, 1, 4);
+
+  layout->addWidget(l_load, 7, 0);
+  layout->addWidget(s_load, 7, 1, 1, 3);
+  layout->addWidget(l_eject, 8, 0);
+  layout->addWidget(s_eject, 8, 1, 1, 3);
+
+  layout->setColumnStretch(0, 2);
+  layout->setColumnStretch(1, 1);
+  layout->setColumnStretch(2, 1);
+  layout->setColumnStretch(3, 2);
+  layout->setRowStretch(10, 10);
+
+  layoutb = new QHBoxLayout();
+  layoutb->setMargin(0);
+  layoutb->setSpacing(3);
+  layoutm->addLayout(layoutb);
+
+  layoutb->addStretch(10);
+  layoutb->addWidget(ck_perm);
+  layoutb->addWidget(pb_save);
+
+  cd = dev->media.type.startsWith("CD");
```

Comment:
Codacy found an issue: [Variable 'cd' is assigned a value that is never used.](https://app.codacy.com/manual/jeremiahwilson777/qpxtool/pullRequest?prid=6191247)